### PR TITLE
GODRIVER-539: Add non panicking BinaryOK function to bson

### PR DESCRIPTION
**Pull request by tobiaskohlbau**
Add method to retrieve binary data without hazard of panicking.

### CODE REVIEW

File: bson/value.go [hunk 1]
```diff
@@ -553,6 +553,16 @@ func (v *Value) Binary() (subtype byte, data []byte) {
 	return st, b
 }
 
+// BinaryOK is the same as Binary, except it returns a boolean instead of
+// panicking.
+func (v *Value) BinaryOK() (subtype byte, data []byte, bool) {
+	if v == nil || v.offset == 0 || v.data == nil || v.data[v.start] != '\x05' {
```
Comment (skriptble) [inline bson/value.go]: Let's use the `TypeBinary` here to make it consistent with the other *OK methods.

File: bson/value.go [hunk 2]
```diff
@@ -553,6 +553,16 @@ func (v *Value) Binary() (subtype byte, data []byte) {
 	return st, b
 }
 
+// BinaryOK is the same as Binary, except it returns a boolean instead of
+// panicking.
+func (v *Value) BinaryOK() (subtype byte, data []byte, bool) {
```
Comment (skriptble) [inline bson/value.go]: There's a compilation issue here, the `bool` parameter needs a name. Changing this to `ok bool`, should work.

File: bson/value.go [hunk 3]
```diff
@@ -553,6 +553,16 @@ func (v *Value) Binary() (subtype byte, data []byte) {
...
+	if v == nil || v.offset == 0 || v.data == nil || v.data[v.start] != '\x05' {
+		return "", nil, false
```
Comment (skriptble) [inline bson/value.go]: This also popped up as a compilation issue, the return here should be a byte, so something like `0x00`.

**Comment by skriptble** [comment]
Hi @tobiaskohlbau,

One more thing, can you add `GODRIVER-539` in the body of the commit message? If you don't have the bandwidth to make these changes let me know and I'll handle it from here.

Thanks for the contribution!
--Kris

**Comment by tobiaskohlbau** [comment]
Hi, thanks for the reply.

If it is okay I would do it tomorrow morning (European time).

File: bson/value.go (hunk 2 above)
Comment (tobiaskohlbau) [reply bson/value.go]: Good catch, sorry for the mistake.

File: bson/value.go (hunk 3 above)
Comment (tobiaskohlbau) [reply bson/value.go]: Again my bad, didn't checked through. Sorry for the inconvenience.

**Comment by tobiaskohlbau** [comment]
Hi @skriptble,

I made the requested changes and added `GODRIVER-539` to the commit body.

Thanks again for your review.

**Pull request merged**